### PR TITLE
Introduce `lua_eval`, `lua_call`, and `sql` grant object types

### LINKED ISSUE #8803: Split execute privilege onto eval,exec and sql
[PRD](https://www.notion.so/tarantool/eval-exec-sql-c8be4c87b3e74a8b9da73d6490d8d6e4)
[Design document](https://www.notion.so/tarantool/New-execute-privileges-677f9a1b115f4f61a945169e48c16dc5)

Now execute privilege affect IPROTO_EVAL and IPROTO_CALL.
We should split it onto two privilege and invent the third privilege that affects calls and evals from SQL.

### PR DESCRIPTION
**Pull request by locker**
In this PR we introduce the new `lua_eval`, `lua_call`, and `sql` object types for [`box.schema.user.grant`](https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_schema/user_grant/) to control access to code execution over the network protocol (IPROTO).

1. Granting the 'execute' privilege on `lua_eval` permits the user to execute arbitrary Lua code with the `IPROTO_EVAL` request.

   Example:

   ```Lua
   box.cfg({listen = 3301})
   box.schema.user.create('alice', {password = 'secret'})
   conn = require('net.box').connect(
       box.cfg.listen, {user = 'alice', password = 'secret'})
   conn:eval('return true') -- access denied
   box.schema.user.grant('alice', 'execute', 'lua_eval')
   conn:eval('return true') -- ok
   ```

2. Granting the 'execute' privilege on `lua_call` permits the user to call any global (accessible via the `_G` Lua table) user-defined Lua function with the `IPROTO_CALL` request. It does **not** permit the user to call built-in Lua functions, such as `loadstring` or `box.session.su`. It does **not** permit the user to call functions registered in the `_func` system space with `box.schema.func.create` (access to those functions is still controlled by privileges granted on `function`).

   Example:

   ```Lua
   function my_func() end
   box.cfg({listen = 3301})
   box.schema.user.create('alice', {password = 'secret'})
   conn = require('net.box').connect(
       box.cfg.listen, {user = 'alice', password = 'secret'})
   conn:call('my_func') -- access denied
   box.schema.user.grant('alice', 'execute', 'lua_call')
   conn:call('my_func') -- ok
   conn:call('box.session.su', {'admin'}) -- access denied
   ```

3. Granting the 'execute' privilege on `sql` permits the user to execute an arbitrary SQL expression with the `IPROTO_PREPARE` and `IPROTO_EXECUTE` requests. Without this privilege or the 'execute' privilege granted on `universe`, the user is **not** permitted to execute SQL expressions over IPROTO anymore. Note that before Tarantool 3.0 any user (even guest) could execute SQL expressions over IPROTO. It is possible to revert to the old behavior by toggling the `sql_priv` compat option.

   Example:

   ```Lua
   box.cfg({listen = 3301})
   box.schema.user.create('alice', {password = 'secret'})
   conn = require('net.box').connect(
       box.cfg.listen, {user = 'alice', password = 'secret'})
   conn:execute('SELECT 1') -- access denied
   box.schema.user.grant('alice', 'execute', 'sql')
   conn:execute('SELECT 1') -- ok
   ```

Closes #8803

### CODE REVIEW

File: changelogs/unreleased/gh-8803-exec-priv.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+* Introduced the new `lua_eval` and `lua_call` object types for
+  `box.schema.user.grant`. Granting the 'execute' privilege on `lua_eval`
+  permits the user to execute an arbitrary Lua expression with the
+  `IPROTO_EVAL` request. Granting the 'execute' privilege on `lua_call`
+  permits the user to execute any global user-defined Lua function with
+  the `IPROTO_CALL` request (gh-8803).
+* **[Breaking change]** Introduced the new `sql` object type for
+  `box.schema.user.grant`. Now, only users that have the 'execute' privilege
+  granted on `sql` or `universe` may execute SQL expressions with the
```
Comment (xuniq) [inline changelogs/unreleased/gh-8803-exec-priv.md:11]:
```suggestion
  `box.schema.user.grant`. Granting the `'execute'` privilege on `lua_eval`
  allows the user to execute an arbitrary Lua expression with the
  `IPROTO_EVAL` request. Granting the `'execute'` privilege on `lua_call`
  allows the user to execute any global user-defined Lua function with
  the `IPROTO_CALL` request (gh-8803).
* **[Breaking change]** Introduced the new `sql` object type for
  `box.schema.user.grant`. Now only users with the 'execute' privilege
  granted on `sql` or `universe` can execute SQL expressions with the
```

**Review by xuniq — APPROVED** [review]

File: changelogs/unreleased/gh-8803-exec-priv.md (hunk 1 above)
Comment (locker) [reply changelogs/unreleased/gh-8803-exec-priv.md]: Fixed. Thanks.

**Review by alyapunov — APPROVED** [review]

**Review by ylobankov — APPROVED** [review]
Tests LGTM

**Comment by locker** [comment]
Fuzzer test fix:

```diff
diff --git a/test/fuzz/xrow_decode_sql_fuzzer.c b/test/fuzz/xrow_decode_sql_fuzzer.c
index 4cdc1e370433..621b0aa12bf7 100644
--- a/test/fuzz/xrow_decode_sql_fuzzer.c
+++ b/test/fuzz/xrow_decode_sql_fuzzer.c
@@ -35,6 +35,7 @@ LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
 	body.iov_len = size;
 
 	struct xrow_header row = {0};
+	row.type = IPROTO_EXECUTE;
 	row.body[0] = body;
 	row.bodycnt = 1;
 
```

**Pull request merged**